spectrum-offchain-cardano/src/data/cfmm_pool.rs

```diff
@@ -700,14 +894,22 @@ impl ApplyOrder<ClassicalOnChainLimitSwap> for ConstFnPool {
             let additional_treasury_y = (((order.base_amount.untag() as u128)
                 * (*self.treasury_fee.numer() as u128))
                 / (*self.treasury_fee.denom() as u128)) as u64;
+            let additional_royalty_y = (((order.base_amount.untag() as u128)
+                * (*self.royalty_fee.numer() as u128))
+                / (*self.royalty_fee.denom() as u128)) as u64;
```

Comment:
Do we have all the necessary validations in pool parser to ensure no division-by-zero is possible?

Reply:
fee_denom is constant and equals to 10000